### PR TITLE
Adds 'empty_password' statement for states.user.present (issue #97)

### PR DESCRIPTION
**Pull request by outime**

### CODE REVIEW

File: users/init.sls [hunk 1]
```diff
@@ -78,9 +78,12 @@ users_{{ name }}_user:
     {% if 'uid' in user -%}
     - uid: {{ user['uid'] }}
     {% endif -%}
-    {% if 'password' in user -%}
+    {% if 'password in user -%}
```
Comment (0xf10e) [inline users/init.sls]: You've introduced a typo here by removing the 2nd ``'``, @outime,

File: users/init.sls [hunk 2]
```diff
@@ -78,9 +78,12 @@ users_{{ name }}_user:
...
     - password: '{{ user['password'] }}'
     {% endif -%}
+    {% if empty_password is defined -%}
```
Comment (0xf10e) [inline users/init.sls]: ``empty_password`` isn't passed into the template as a context variable, it's a key of ``user`` thus line 84 would never evaluate to true.
To use ``'empty_password' in user`` is the right way to test for a key's existence.

**Comment by 0xf10e** [comment]
Sorry, I didn't realize you've made two updates to #97 so I didn't read all your code snippets close enough. Could have caught those two earlier.

**Comment by outime** [comment]
@0xf10e thanks for the heads up! Hope I got it right now.

File: users/init.sls [hunk 3]
```diff
@@ -81,6 +81,9 @@ users_{{ name }}_user:
     {% if 'password' in user -%}
     - password: '{{ user['password'] }}'
     {% endif -%}
+    {% if 'empty_password' in user -%}
```
Comment (puneetk) [inline users/init.sls]: Can you change this to
```
 {% if user.get('empty_password') %} 
- empty_password: {{ user.get('empty_password') }}
{% endif -%}
```

**Comment by outime** [comment]
@puneetk :+1:

**Pull request merged**